app/src/full/java/io/homeassistant/companion/android/sensors/LocationSensorManager.kt

```diff
@@ -765,22 +766,17 @@ class LocationSensorManager : LocationSensorManagerBase() {
         lastLocationSend = now
         lastUpdateLocation = updateLocation.gps.contentToString()
 
-        val geoSensorManager = SensorReceiver.MANAGERS.firstOrNull { it.getAvailableSensors(latestContext).any { s -> s.name == commonR.string.basic_sensor_name_geolocation } }
-        val geoSensor = AppDatabase.getInstance(latestContext).sensorDao().getFull(GeocodeSensorManager.geocodedLocation.id)
-
         ioScope.launch {
             try {
                 integrationUseCase.updateLocation(updateLocation)
                 Log.d(TAG, "Location update sent successfully")
 
                 // Update Geocoded Location Sensor
-                SensorReceiver().updateSensor(
-                    latestContext,
-                    integrationUseCase,
-                    geoSensor,
-                    geoSensorManager,
-                    GeocodeSensorManager.geocodedLocation
-                )
+                val intent = Intent(latestContext, SensorReceiverBase::class.java)
```

Comment:
`SensorReceiverBase` isn't annotated with `@AndroidEntryPoint` so it this intent won't trigger injection and doesn't solve the crash. Using the base is like an implementation detail, so I think the app should send it to `SensorReceiver` instead (which is annotated).